### PR TITLE
feat(autodiscover): allow array for autodiscoverfilter

### PR DESCRIPTION
**Pull request by secustor**
## Changes:
Change `autodiscoverFilter` to array.  

Introduce massaging after file extraction. 

## Context:
Part of: 
- https://github.com/renovatebot/renovate/issues/8763

Reimplements: 
- https://github.com/renovatebot/renovate/pull/9453
- https://github.com/renovatebot/renovate/pull/9886

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

Without preset: https://github.com/secustor/renovate_autodiscoverfilter
With preset: https://github.com/secustor/renovatebot-golangtest

### CODE REVIEW

File: lib/workers/global/autodiscover.ts [hunk 1]
```diff
@@ -30,7 +30,13 @@ export async function autodiscoverRepositories(
     return config;
   }
   if (config.autodiscoverFilter) {
-    discovered = discovered.filter(minimatch.filter(config.autodiscoverFilter));
+    const matched = new Set<string>();
+    for (const filter of config.autodiscoverFilter) {
+      const res = minimatch.match(discovered, filter);
+      res.forEach((e) => matched.add(e));
```
Comment (viceice) [inline lib/workers/global/autodiscover.ts:36]: please use a for-of loop

**Review by viceice — COMMENTED** [review]
needs env and cli migration too.

**Comment by secustor** [comment]
Tested this with: 

renovate.json:
```
{
  autodiscoverFilter: "secustor/renovate*"
}
```
```
{
  autodiscoverFilter: [
      "secustor/renovate*",
      "secustor/test*"
  ]
}
```

cli args:
`--autodiscover-filter="secustor/renovate*"`
`--autodiscover-filter=secustor/renovate*`
`--autodiscover-filter=secustor/renovate*,secustor/test*"`

env: 
`RENOVATE_AUTODISCOVER_FILTER=secustor/renovate*,secustor/test*`  
`RENOVATE_AUTODISCOVER_FILTER=secustor/renovate*`
`RENOVATE_CONFIG={"autodiscoverFilter": ["secustor/renovate*","secustor/test*"]}`
`RENOVATE_CONFIG={"autodiscoverFilter": "secustor/renovate*"}`

**Review by viceice — APPROVED** [review]
Hopefully we don't get new bad side-effects because of the new config massaging

**Comment by cascornelissen** [comment]
This seems to be breaking our setup where we define the filter similar to this:

```
export RENOVATE_AUTODISCOVER_FILTER="project/{a,b,c,d,*-app}"
```

This gets mapped to the following value in 31.2.2

```
"autodiscoverFilter": "project/{a,b,c,d,*-app}"
```

But it breaks to the following after that.

```
"autodiscoverFilter": [
    "project/{a",
    "b",
    "c",
    "d",
    "*-app}"
]
```

**Pull request merged**